qcengine/programs/qchem.py

```diff
@@ -0,0 +1,254 @@
+        return exe_success, proc
+
+    def build_input(self, input_model: 'ResultInput', config: 'JobConfig',
+                    template: Optional[str] = None) -> Dict[str, Any]:
+
+        # Check some bounds on what cannot be parsed
+        if "ccsd" in input_model.method.lower() or "ccd" in input_model.method.lower():
+            raise InputError("Cannot handle CC* methods currently.")
+
+        # Build keywords
+        keywords = {k.upper(): v for k, v in input_model.keywords.items()}
+        keywords["INPUT_BOHR"] = "TRUE"
+        keywords["MEM_TOTAL"] = str(int(config.memory * 1024))  # In MB
+
+        if input_model.driver == "energy":
+            keywords["JOBTYPE"] = "sp"
+        elif input_model.driver == "gradient":
+            keywords["JOBTYPE"] = "force"
+        elif input_model.driver == "hessian":
+            keywords["JOBTYPE"] = "freq"
+        else:
+            raise InputError(f"Driver of type {input_model.driver} is not yet supported.")
+
+        if input_model.molecule.fix_com or input_model.molecule.fix_orientation:
+            keywords["NO_REORIENT"] = "TRUE"
+
+        keywords["METHOD"] = input_model.model.method
+        if input_model.model.basis:
+            keywords["BASIS"] = input_model.model.basis
+
+        # Begin the input file
+        input_file = []
+        input_file.append(f"""$comment
+Automatically generated Q-Chem input file by QCEngine
+$end
+            """)
+
+        # Add Molecule, TODO: Add to QCElemental
+        mol = input_model.molecule
+        input_file.append("$molecule")
+        input_file.append(f"""{int(mol.molecular_charge)} {mol.molecular_multiplicity}""")
+
+        for real, sym, geom in zip(mol.real, mol.symbols, mol.geometry):
+            if real is False:
+                raise InputError("Cannot handle ghost atoms yet.")
+            input_file.append(f"{sym} {geom[0]:14.8f} {geom[1]:14.8f} {geom[2]:14.8f}")
+
+        input_file.append("$end\n")
+
+        # Write out the keywords
+        input_file.append("$rem")
+        for k, v in keywords.items():
+            input_file.append(f"{k:20s}  {v}")
+        input_file.append("$end\n")
+
+        ret = {
+            "infiles": {
+                "dispatch.in": "\n".join(input_file)
+            },
+            "commands": [which("qchem"), "-nt",
+                         str(config.ncores), "dispatch.in", "dispatch.out"],
+            "scratch_directory": config.scratch_directory,
+        }
+
+        return ret
+
+    def parse_output(self, outfiles: Dict[str, str], input_model: 'ResultInput') -> 'Result':
+
+        output_data = {}
+        outfiles["dispatch.out"]
+
+        bdata = {}
+        for k, v in outfiles.items():
+            if k == "dispatch.out":
+                continue
+            if v is None:
+                continue
+            bdata[k] = np.frombuffer(v)
+
+        if input_model.driver == "energy":
+            output_data["return_result"] = bdata["99.0"][-1]
```

Comment:
solvation, xdm, excited states. this looks frighteningly simple in the face of those complications. but i'll believe it holds for a good swath.